DataConnectors/JumpCloud Single Sign On/AzureFunctionJumpCloud/JCQueueTrigger1/run.ps1

```diff
@@ -0,0 +1,141 @@
+    Version:        1.0.1
+    Author(s):      Microsoft - Chris Abberley
+    Last Modified:  2020-08-03
+    Comment:        First Release
+
+    DESCRIPTION
+    This Function App calls the JumpCloud Directory Insights API (https://jumpcloud-insights.api-docs.io/1.0/api-overview/directory-insights) to pull the JumpCloud logs. The response from the JumpCloud API is recieved in JSON format. This function will build the signature and authorization header 
+    needed to post the data to the Log Analytics workspace via the HTTP Data Connector API. The Function App will post the JumpCloud logs to the JumpCloud_CL table in the Log Analytics workspace.
+#>
+
+# Input bindings are passed in via param block.
+param([string] $QueueItem, $TriggerMetadata)
+
+# Write out the queue message and insertion time to the information log.
+Write-Output "JumpCloud: Queue trigger for work item: $QueueItem, Queue item insertion time: $($TriggerMetadata.InsertionTime)"
+
+import-module AzTable
+
+# Retrieve Environment Variables and prep other Variables for the JumpCloud API request
+$JCService = $QueueItem                         #Which eventlog set to rerieve from JumpCloud
+$JCapiToken = $env:JumpCloudApiToken            #JumpCloud API
+$JCuri = $env:JumpCloudUri                      #Standard JumpCloud URI
+#$JCStartTime = $env:JumpCloudStartTime          #Initial Start time to collect logs
+$AzureWebJobsStorage =$env:AzureWebJobsStorage  #Storage Account to use for table to maintain state for log queries between executions
+$customerId = $env:workspaceId                  #Log Analytics Details
+$sharedKey =  $env:workspaceKey                 #Log Analytics Details
+$JCTablename = $env:AzureSentinelTable          #"JumpCloudQTest" #LogAnalytics Tablename which will have '_CL' added to it by LogAnalytics
+$TimeStampField = "timestamp"                   #define JumpCloud Timestamp field
+$JCSearchAfter = ""
+$totalrecordcount =0
+
+#fix date format
+$DateFix = Get-Date
+$JCStartTime = $DateFix.ToString('yyyy-MM-ddT00:00:00Z')
+# using Azure Table on Functions Azure Storage to maintain state between runs
+# Retrieve JumpCloud last retrieved record and Date-time or if first run create table
+$JCstorage =  New-AzStorageContext -ConnectionString $AzureWebJobsStorage
+$StorageTable = Get-AzStorageTable -Name $JCTablename -Context $JCStorage -ErrorAction Ignore
+if($null -eq $storageTable.Name){  
+    $result = New-AzStorageTable -Name $JCTablename -Context $JCstorage
+    $JCTable = (Get-AzStorageTable -Name $JCTablename -Context $JCstorage.Context).cloudTable
+    $result = Add-AzTableRow -table $JCTable -PartitionKey $JCapiToken -RowKey $JCService -property @{'SearchAfter' = "'"+$JCSearchAfter+"'";"StartTime"=$JCStartTime} -UpdateExisting
+}
+Else {
+    $JCTable = (Get-AzStorageTable -Name $JCTablename -Context $JCstorage.Context).cloudTable
+}
+$row = Get-azTableRow -table $JCTable -partitionKey $JCapiToken -RowKey $JCService -ErrorAction Ignore
```

Comment:
Not sure the reason behind this. But usually api key (secret) should not be the partition key of the table.